src/Kovan-DssSpell.sol

```diff
@@ -16,39 +16,284 @@
+}
+
+interface Bumpable {
+    function bump(bytes32, uint256) external;
+}
 
 contract DssSpellAction is DssAction {
 
-    string public constant description = "Kovan Spell";
+    // Provides a descriptive tag for bot consumption
+    // This should be modified weekly to provide a summary of the actions
+    // Hash: seth keccak -- "$(wget https://raw.githubusercontent.com/makerdao/community/2a7a8c915695b7298fe725ee3dc6c613fa9d9bbe/governance/votes/Executive%20vote%20-%20April%2012%2C%202021.md -q -O - 2>/dev/null)"
+    string public constant description = "RWA003-RWA006 integration";
 
-    // Turn off office hours
-    function officeHours() public override returns (bool) {
-        return false;
-    }
+    // Many of the settings that change weekly rely on the rate accumulator
+    // described at https://docs.makerdao.com/smart-contract-modules/rates-module
+    // To check this yourself, use the following rate calculation (example 8%):
+    //
+    // $ bc -l <<< 'scale=27; e( l(1.08)/(60 * 60 * 24 * 365) )'
+    //
+    // A table of rates can be found at
+    //    https://ipfs.io/ipfs/QmefQMseb3AiTapiAKKexdKHig8wroKuZbmLtPLv4u2YwW
+    //
+    uint256 constant TWO_PCT            = 1000000000627937192491029810;
+    uint256 constant FOUR_PT_FIVE_PCT   = 1000000001395766281313196627;
+    uint256 constant SIX_PCT            = 1000000001847694957439350562;
+    uint256 constant SEVEN_PCT          = 1000000002145441671308778766;
 
-    uint256 constant WAD = 10**18;
-    uint256 constant RAY = 10**27;
-    uint256 constant RAD = 10**45;
+    // Math
+    uint256 constant THOUSAND = 10 ** 3;
+    uint256 constant MILLION  = 10 ** 6;
+    uint256 constant WAD      = 10 ** 18;
+    uint256 constant RAY      = 10 ** 27;
+    uint256 constant RAD      = 10 ** 45;
 
-    address constant MCD_FLASH = 0x5aA1323f61D679E52a90120DFDA2ed1A76E4475A;
+    // Maker changelog 
+    address public constant MAKER_CHANGELOG = 0xdA0Ab1e0017DEbCd72Be8599041a2aa3bA7e740F;
 
     function actions() public override {
-        // ---------------------------- Add MCD_FLASH ----------------------------
-        address MCD_VAT = DssExecLib.vat();
-        Fileable(MCD_FLASH).file("max", 500_000_000 * WAD);
-        Fileable(MCD_FLASH).file("toll", 5 * WAD / 10000);
-        DssExecLib.authorize(MCD_VAT, MCD_FLASH);
-        DssExecLib.setChangelogAddress("MCD_FLASH", MCD_FLASH);
-
-        // ---------------------------- Update Chainlog version ----------------------------
-        DssExecLib.setChangelogVersion("1.9.1");
+        CentrifugeCollateralValues memory RWA003 = CentrifugeCollateralValues({
+            MCD_JOIN: 0x4CCc7fED3912A32B6Cf7Db2FdA1554a9FF574099,
+            GEM: 0xDBC559F5058E593981C48f4f09fA34323df42d51,
+            OPERATOR: 0x45e17E350279a2f28243983053B634897BA03b64,
+            INPUT_CONDUIT: 0x45e17E350279a2f28243983053B634897BA03b64,
+            OUTPUT_CONDUIT: 0x45e17E350279a2f28243983053B634897BA03b64,
```

Comment:
The `urn` on the manager is `0xc615F4188C255445290fB9E6dB5E021fe4CA8ECf`.  This is `RWA002_A_URN`.  You need to update the manager to have `0x993c239179D6858769996bcAb5989ab2DF75913F` as the `urn`.  Looks like the deployer denied itself as a `ward`.  Do you guys still have permissions to fix this?

Reply:
We recently added the deployment of the manager to our deployment scripts, and initially accidentally also added the urn wiring. We realized later that the `urn` cannot yet have been deployed on deployment of the manager, since mip21 requires the manager, so it's a bit of a chicken and egg problem. Luckily, we still have access to this, through our kovan admin account (`0x0A735602a357802f553113F5831FE2fbf2F0E2e0`), which is relied on all the ROOT contracts (`0x792164b3e10a3CE1efafF7728961aD506c433c18` for RWA003 which you mention above). We'll fix all of the urn wirings today.

On mainnet, the urn hasn't yet been filed on the managers, and we'll do all of this in one spell after the executive vote has passed, combined with the locking of the RWA token and some other integration tasks.

Reply:
This is now fixed.